docs/tutorials/pi/make-a-plant-watering-robot.md

```diff
@@ -0,0 +1,483 @@
+
+{{< tabs >}}
+{{% tab name="Config Builder" %}}
+
+<img src="../../img/plant-watering-pi/sensor-process-config-builder.png" alt="Creation of a pump motor in the Viam app config builder." style="max-width:800px" />
+
+{{% /tab %}}
+{{% tab name="JSON Template" %}}
+
+```json {class="line-numbers linkable-line-numbers"}
+{
+  // "components": [ board & motor ... ] ,
+  // "remotes" [ my-sensor-server ... ] ,
+  "processes": [
+    {
+      "id": "run-sensor-server",
+      "log": true,
+      "name": "sudo",
+      "args": [
+        "-u",
+        "<my_username>",
+        "python",
+        "sensor.py"
+      ],
+      "cwd": "/home/<my_username>/plant-watering-robot"
+    }
+  ]
+}
+```
+
+{{% /tab %}}
+{{< /tabs >}}
+
+### Add Python Control Code
+
+Follow these instructions to start working on your Python Control Code:
+
+1. Navigate to your robot's page in [the Viam app](https://app.viam.com), and click on the **CODE SAMPLE** tab.
+Follow the instructions in this tab.
+2. Click **COPY CODE** to copy a code sample that establishes a connection with your robot when run.
+3. Paste this code sample into a new file in your `plant-watering-robot` directory.
+4. Name the file <file>plant-watering-robot.py</file>, and save it.
+  
+For example, run the following commands to create and open the file:
+
+``` shell
+cd plant-watering-robot
+touch plant-watering-robot.py
+vim plant-watering-robot.py
+```
+
+Now, you can add code into <file>plant-watering-robot.py</file> to write the logic that defines your plant watering system.
+
+<img src="../../img/plant-watering-pi/main-py.png" alt="The main.py function from your robot's Code Sample tab, ready to edit." style="max-width:800px" />
+
+<br>
+</br>
+
+Use the Viam [motor](/components/motor#api) and [sensor](/components/sensor#control-your-sensor-with-viams-client-sdk-libraries) API methods.
+Determine at what [analog value from the Soil Moisture readings](#test-your-soil-moisture-readings-on-your-pi) you want to water your plant.
+Also, consider how often you would like to check the moisture levels of the plant, and how long the plant should be watered.
+
+You can get your components from the robot like this:
+
+```python
+sensor = Sensor.from_robot(robot = robot, name = 'moisture_sensor') # Note that this name, `moisture_sensor`, is defined in sensor.py
+water_pump = Motor.from_robot(robot = robot, name = 'water_pump')
+```
+
+And you can add your system logic to run continuously like this:
+
+```python
+ while True:
+
+    # Get moisture sensor readings
+    readings = await sensor.get_readings()
+    print(readings)
+    avg = sum(readings) / len(readings)
+    sleep(1)
+
+    if(avg > 950):
+    print('needs water')
+
+    # Run for 20 seconds
+    await water_pump.go_for(1000,200)
+
+    # Wait for the water to soak in
+    sleep(5)
+
+    # Recheck the avg
+    readings = await sensor.get_readings()
+    print(readings)
+    avg = sum(readings) / len(readings)
+    sleep(1)
+```
+
```

Comment:
Can you add the full code at the end here in case folks get lost? And link to it from the beginning of the tutorial?

Reply:
@npentrel Olivia did not provide full code, and the full code would differ per-robot, so I'm not sure if this would be reasonable-- @mcvella what do you think? 

I would normally be inclined to add. Maybe just a full `main()` function?

Reply:
Usually we'd have code in a repo somewhere unless its very small, in which case it could be embedded in the tutorial.  I think readers appreciate being able to just copy/paste (and add credentials) and be off and running.

Reply:
but main function at minimum would be better I think

Reply:
I suppose you got it to fully work with the robot so you could put that in a gist (gist.github.com) and link that?